library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -34,6 +34,7 @@
     private static final String[] PROJECT_BATCH_ID = {DownloadContract.Batches._ID};
     private static final String WHERE_DELETED_VALUE_IS = DownloadContract.Batches.COLUMN_DELETED + " = ?";
     private static final String[] MARKED_FOR_DELETION = {"1"};
+    private static final String SORT_BY_ID_DESCENDING = DownloadContract.Batches._ID + " DESC";
```

Comment:
no and yes, until we have a proper queue order column. The problem we're going to suffer with is running checks on the queue, should we block all downloads if 1 item in the queue has been denied?